packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummary.ts

```diff
@@ -0,0 +1,252 @@
+import { BigNumber, constants, utils } from 'ethers'
+import { useAccount } from 'wagmi'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { useAppState } from '../../state'
+import { useGasPrice } from '../useGasPrice'
+import { useDebouncedValue } from '../useDebouncedValue'
+import {
+  isTokenArbitrumSepoliaNativeUSDC,
+  isTokenArbitrumOneNativeUSDC
+} from '../../util/TokenUtils'
+import { withdrawInitTxEstimateGas } from '../../util/WithdrawalUtils'
+import { depositEthEstimateGas } from '../../util/EthDepositUtils'
+import { depositTokenEstimateGas } from '../../util/TokenDepositUtils'
+import { useNetworksRelationship } from '../useNetworksRelationship'
+import { useNetworks } from '../useNetworks'
+import { useArbQueryParams } from '../useArbQueryParams'
+import { useNativeCurrency } from '../useNativeCurrency'
+
+const INITIAL_GAS_ESTIMATION_RESULT: GasEstimationResult = {
+  // Estimated L1 gas, denominated in Wei, represented as a BigNumber
+  estimatedL1Gas: constants.Zero,
+  // Estimated L2 gas, denominated in Wei, represented as a BigNumber
+  estimatedL2Gas: constants.Zero,
+  // Estimated L2 submission cost is precalculated and includes gas price
+  estimatedL2SubmissionCost: constants.Zero
+}
+
+const INITIAL_GAS_SUMMARY_RESULT: UseGasSummaryResult = {
+  estimatedL1GasFees: 0,
+  estimatedL2GasFees: 0
+}
+
+export type GasEstimationStatus =
+  | 'loading'
+  | 'success'
+  | 'error'
+  | 'unavailable'
+
+export type GasEstimationResult = {
+  estimatedL1Gas: BigNumber
+  estimatedL2Gas: BigNumber
+  estimatedL2SubmissionCost: BigNumber
+}
+
+export type UseGasSummaryResult = {
+  estimatedL1GasFees: number
+  estimatedL2GasFees: number
+}
+
+export function useGasSummary(): {
+  gasSummary: UseGasSummaryResult
+  gasSummaryStatus: GasEstimationStatus
+} {
+  const {
+    app: { arbTokenBridge, arbTokenBridgeLoaded, selectedToken: token }
+  } = useAppState()
+  const [networks] = useNetworks()
+  const {
+    childChain,
+    childChainProvider,
+    parentChain,
+    parentChainProvider,
+    isDepositMode
+  } = useNetworksRelationship(networks)
+  const { address: walletAddress } = useAccount()
+  const [{ amount }] = useArbQueryParams()
+  const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
+  const [gasSummaryStatus, setGasSummaryStatus] =
+    useState<GasEstimationStatus>('loading')
+  const [gasSummary, setGasSummary] = useState<UseGasSummaryResult>(
+    INITIAL_GAS_SUMMARY_RESULT
+  )
```

Comment:
Let's just merge the two together, that will allow us to get rid of the extra useEffect

Reply:
address here https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1490